operator/pkg/client/users/client.go

```diff
@@ -173,11 +176,19 @@ func (c *Client) getPassword(ctx context.Context, user *redpandav1alpha2.User) (
 				return "", err
 			}
 
+			if ptr.Deref(auth.Password.NoGenerate, false) {
```

Comment:
Rather than making a new error type, could we update the check on 175 to take `NoGenerate` into consideration?

Reply:
Agree done.